lib/internal/url.js

```diff
@@ -85,34 +85,36 @@ class TupleOrigin {
   }
 }
 
+// Resued by URL constructor and
+// URL#href setter.
+function parse(input, base) {
+  input = String(input);
+  const base_context = base ? base[context] : undefined;
+  this[context] = new StorageObject();
+  binding.parse(input.trim(), -1, base_context, undefined,
+    (flags, protocol, username, password,
+      host, port, path, query, fragment) => {
+      if (flags & binding.URL_FLAGS_FAILED)
+        throw new TypeError('Invalid URL');
+      this[context].flags = flags;
+      this[context].scheme = protocol;
+      this[context].username = username;
+      this[context].password = password;
+      this[context].port = port;
+      this[context].path = path;
+      this[context].query = query;
+      this[context].fragment = fragment;
+      this[context].host = host;
+      this[searchParams] = new URLSearchParams(query);
```

Comment:
The [`searchParams` attribute](https://url.spec.whatwg.org/#dfnReturnLink-0) is declared with `[SameObject]`. When the `href` setter is invoked, the underlying header list should be replaced, but not the URLSearchParams object itself.